src/plugins/recursion/README.md

```diff
@@ -10,18 +10,37 @@ Also I have no idea what your `struct` plugin actually does. There is no example
 the contract which doesn't help understanding the plugin.
 
 ## Problems
-1) We want to check if all recursions are non-cyclical
-2) We want to check if a recursive config actually leads to another correct config and not into NULL for example
-3) Forcing users to define non-recursive settings first on which they can build up 
-recursive settings is inconvenient and could lead to orphan settings.
-4) We can only check right after calling `kdb set` and not a whole config. 
-This means that user can have "unfinished" configurations while setting new values.
+1) We want to check if all recursion links are non-cyclical, eg. `a` -> `b` -> `c` -> `a`
+2) Imagine a structure like `a` -> `b`. A user sets `a` but forgets `b`. He links `a` to (yet non existent) `b`.
+Now there are two possibilties. Either we prohibit the linking and emit an error, forcing the user to define
+the `b` setting first or we just emit a warning and expect the user to declare `b` later. The latter one could lead to
+incomplete configurations.
+
+The first option would require a certain ordering which clashes with the fact that some storage plugins do not
+premain the ordering. An example:
+```
+[b]
+
+[a]
+ref = b
+```
+
+is a legal configuration. Loading and saving the configuration again could lead to a reversed structure
+
+```
+[a]
+ref = b
+
+[b]
+```
+
+which could cause an error since `a` cannot be loaded as `b` is yet non existent.
```

Comment:
Can you explain why? How exactly are keys loaded into the kdb?

I assume that it is loaded in the order of the file, i.e. it would be the same if I call kdb set ... in the same order as the config file

Reply:
The plugins receive a whole keyset and can do whatever they want to do with it. So in the implementation you can lookup if something is present, it does not matter where in the keyset it is.